core/src/services/azdls/core.rs

```diff
@@ -97,17 +96,17 @@ impl AzdlsCore {
 }
 
 impl AzdlsCore {
-    pub async fn azdls_read(&self, path: &str, range: BytesRange) -> Result<Response<HttpBody>> {
-        let p = build_abs_path(&self.root, path);
-
-        let url = format!(
+    fn build_path_url(&self, path: &str) -> String {
+        format!(
             "{}/{}/{}",
             self.endpoint,
             self.filesystem,
-            percent_encode_path(&p)
-        );
+            percent_encode_path(build_abs_path(&self.root, path).trim_end_matches('/'))
+        )
+    }
 
-        let mut req = Request::get(&url);
+    pub async fn azdls_read(&self, path: &str, range: BytesRange) -> Result<Response<HttpBody>> {
+        let mut req = Request::get(self.build_path_url(path));
```

Comment:
Please revert unrelated changes.

Reply:
I guess what you mean is just a display bug in the GitHub diff?